nncf/experimental/onnx/statistics/collectors.py

```diff
@@ -0,0 +1,77 @@
+from typing import Union, List, Deque
+
+import numpy as np
+
+from nncf.common.tensor import NNCFTensor
+from nncf.common.tensor import TensorElementsType
+from nncf.common.tensor_statistics.collectors import MinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import NNCFCollectorTensorProcessor
+from nncf.common.tensor_statistics.collectors import MeanMinMaxStatisticCollector
+from nncf.experimental.onnx.tensor import ONNXNNCFTensor
+from nncf.experimental.onnx.statistics.statistics import ONNXMinMaxTensorStatistic
+
+
+class ONNXNNCFCollectorTensorProcessor(NNCFCollectorTensorProcessor):
+    """
+    """
+
+    @staticmethod
+    def reduce_min(x: NNCFTensor, axis: Union[int, tuple]) -> NNCFTensor:
+        return ONNXNNCFTensor(np.amin(x.tensor, axis=axis))
+
+    @staticmethod
+    def reduce_max(x: NNCFTensor, axis: Union[int, tuple]) -> NNCFTensor:
+        return ONNXNNCFTensor(np.amax(x.tensor, axis=axis))
+
+    @staticmethod
+    def abs(x: NNCFTensor) -> NNCFTensor:
+        return ONNXNNCFTensor(np.abs(x.tensor))
+
+    @staticmethod
+    def min(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
+        return ONNXNNCFTensor(np.min(x1.tensor, x2.tensor))
+
+    @staticmethod
+    def max(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
+        return ONNXNNCFTensor(np.max(x1.tensor, x2.tensor))
+
+    @staticmethod
+    def mean(x: NNCFTensor, axis: Union[int, tuple]) -> NNCFTensor:
+        return ONNXNNCFTensor(np.mean(x.tensor, axis=axis))
+
+    @staticmethod
+    def stack(x: Union[List[NNCFTensor], Deque[NNCFTensor]], axis: int = 0) -> NNCFTensor:
+        x = [t.tensor for t in x]
+        return ONNXNNCFTensor(np.stack(x, axis=axis))
+
+    @staticmethod
+    def unstack(x: NNCFTensor, axis: int = 0) -> List[NNCFTensor]:
+        return np.split(x, axis=axis)
+
+    @staticmethod
+    def sum(tensor: NNCFTensor) -> TensorElementsType:
+        return np.sum(tensor.tensor)
+
+
+class ONNXMinMaxStatisticCollector(MinMaxStatisticCollector):
```

Comment:
We also need to handle the case when statistics aggregators are represented by subgraphs that are inserted by the statistics collector. It should be done in a separate PR. But for now, I suggest thinking about the object that can encapsulate the logic of "in-place" statistics (subgraph) and statistics aggregators (that compute statistics afterward). Each backend (ONNX, IR, etc.) should implement interfaces for statistics aggregators and "in-place" statistics if the latter is supported. So, the statistics collector can switch between these two approaches depending on the supported and selected options.

Reply:
I think that the current interface of StatisticsAggregator should be fine. 
Also, take notice that now statistics calculation is done out of the algorithms. It is done by CompressionBuilder, which creates StatisticsAggregator.
Are you satisfied with that?